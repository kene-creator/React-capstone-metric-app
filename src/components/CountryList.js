/* eslint-disable react/prop-types */
import React from 'react';

const CountryList = (props) => {
  const { data } = props;

  return (
    <ul className="flex flex-wrap">
      {data.map((countryData, i) => (
        <li
          key={countryData.id}
          className={`basis-2/4 flex justify-center items-center flex-col border-b border-white ${
            i % 2 === 0 ? 'bg-[#DC4782]' : ''
          } ${i % 2 !== 0 ? 'bg-[#CF4278] border-l border-white' : ''}`}
        >
          <div>
            <img
              src={countryData.image}
              alt="Country flag"
              className="w-16 h-16 rounded-full mt-4 drop-shadow-lg"
            />
          </div>
          <div className="flex items-center justify-center">
            <h3 className="text-white mt-4 mb-2 font-semibold text-center">
              {countryData.name}
            </h3>
          </div>
        </li>
      ))}
    </ul>
  );
};

export default CountryList;
