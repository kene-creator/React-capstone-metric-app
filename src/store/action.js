import axios from 'axios';
import { createAsyncThunk } from '@reduxjs/toolkit';

const FETCH_COUNTRY = 'country-name/country/FETCH_COUNTRY';

const API_URL = 'https://restcountries.com/v3.1/region/europe';

const countryList = [];

const countryReducer = (state = countryList, action) => {
  const { type, payload } = action;
  console.log(type, payload);
  switch (type) {
    case 'country-name/country/FETCH_COUNTRY/fulfilled':
      return payload.country;

    default:
      return state;
  }
};

export const fetchCountry = createAsyncThunk(FETCH_COUNTRY, async () => {
  const res = await axios.get(API_URL);
  const resultArray = res.data;
  console.log(resultArray);

  return {
    // eslint-disable-next-line no-unused-vars
    country: resultArray.map((value) => ({
      id: value.area,
      population: value.population,
      name: value.name.official,
      languages: value.languages,
      capital: value.capital,
      image: value.flags.png,
      timezone: value.timezones[0],
      subregion: value.subregion,
    })),
  };
});

export default countryReducer;
