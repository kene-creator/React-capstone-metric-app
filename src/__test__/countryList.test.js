import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../store/store';
import CountryList from '../../__mock__/countryList';

describe('App components snapshots', () => {
  it('Country List renders correctly', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <CountryList />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
