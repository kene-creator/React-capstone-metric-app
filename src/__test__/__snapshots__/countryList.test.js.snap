// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App components snapshots Country List renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-wrap"
      >
        <a
          class="basis-2/4 flex justify-center items-center flex-col border-b border-white bg-[#DC4782] "
          href="/dataPage/23"
        >
          <div>
            <img
              alt="Country flag"
              class="w-16 h-16 rounded-full mt-4 drop-shadow-lg my-auto mx-auto"
              src="https://flagcdn.com/w320/se.png"
            />
          </div>
          <div
            class="flex items-center justify-center"
          >
            <h3
              class="text-white mt-4 mb-2 font-semibold text-center"
            />
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-wrap"
    >
      <a
        class="basis-2/4 flex justify-center items-center flex-col border-b border-white bg-[#DC4782] "
        href="/dataPage/23"
      >
        <div>
          <img
            alt="Country flag"
            class="w-16 h-16 rounded-full mt-4 drop-shadow-lg my-auto mx-auto"
            src="https://flagcdn.com/w320/se.png"
          />
        </div>
        <div
          class="flex items-center justify-center"
        >
          <h3
            class="text-white mt-4 mb-2 font-semibold text-center"
          />
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
